### PR TITLE
[SELS-TASK][BE] Sign In Functions

### PR DESCRIPTION
**Pull request by michaelsilverio**
### [Asana Link]
https://app.asana.com/0/1201713838250812/1201726038594789/f

### [Commands]
```
# after modifying your env file
php artisan migrate

# makes new user
php artisan db:seed

php artisan serve
npm run watch
```

### [Pre-conditions]
After running `npm run watch` and `php artisan serve`. Go to the url being generated on artisan serve

// Sign In Details
email: user@test.com | password: test

### [Expected Output]
Details

- Button submit is disabled on first render
- Validation message on Email input
- Validation message on Password input
- If all fields are not empty, the button submit is Enabled.
- On submit, if Unknown user will display "Wrong Email or Password"
- On submit successful, will output a dialog with "Signing In..." message. Will redirect to dashboard after 2.5 seconds

### [Notes]
- No redirect if user is already authenticated ( for future improvement )
- Remember Me checkbox is not functional ( for future improvement )
- Forgot Password is not functional ( for future improvement )
- Sign Up Button Not Functional

### [Screenshots]
none

### CODE REVIEW

File: resources/js/actions/userAuthActions.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    var data = {};
+    var submitDisabled = true;
+    const email = formData.get('email');
+    const password = formData.get('password');
+    const users = {
+        email: email,
+        password: password
+    }
+    // if all fields are not blank 
+    if (users.email != '' && users.password != '') {
```
Comment (AubreyV) [inline resources/js/actions/userAuthActions.js:14]: use strict equality/inequality operators

```suggestion
    if (users.email !== '' && users.password !== '') {
```

File: resources/js/actions/userAuthActions.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        await userApi.post('/sign-in', users)
+        .then(response => {
+            // this is used for error checking display
+            data = {
+                submitDisabled: false,
+                errors: response.data.errors,
+                message: response.data.message
+            }
+        })
+    }else if (users.email == '' && users.password != '') {
```
Comment (AubreyV) [inline resources/js/actions/userAuthActions.js:24]: same here apply strict inequality opetator, and put space after `}`

```suggestion
    } else if (users.email === '' && users.password !== '') {
```

File: resources/js/actions/userAuthActions.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        data = { submitDisabled ,email: 'Email should not be empty' }
+    }else if (users.email != ''  && users.password == '') {
```
Comment (AubreyV) [inline resources/js/actions/userAuthActions.js:26]: same here apply strict inequality opetator, and put space after `}`

```suggestion
    } else if (users.email !== ''  && users.password === '') {
```

File: resources/js/actions/userAuthActions.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    });
+}
+
+// check if in email format
+export const validateEmail = email => dispatch => {
+    var submitDisabled = true;
+    var check = /\S+@\S+\.\S+/;
+    const validation = check.test(email) == true ? '' : 'Invalid Email';
+
+    if (validation == true) {
```
Comment (AubreyV) [inline resources/js/actions/userAuthActions.js:48]: validation value from line 46 is neither true or false, it's a string. Please check this again.

File: resources/js/actions/userAuthActions.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        }
+    }
+
+    dispatch({
+        type: SIGN_IN,
+        payload: data
+    });
+}
+
+// check if in email format
```
Comment (AubreyV) [inline resources/js/actions/userAuthActions.js:42]: Read about [How to write commnts in JS](https://www.digitalocean.com/community/tutorials/how-to-write-comments-in-javascript) and apply it here.

File: resources/js/actions/userAuthActions.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        }
+    });
+}
+
+// check if password is blank
+export const validatePassword = password => dispatch => {
+    var submitDisabled = true;
+    const validation = password != '' ? '' : 'Password should not be empty';
+
+    if (validation == '') {
```
Comment (AubreyV) [inline resources/js/actions/userAuthActions.js:66]: apply strict equality opetator

File: resources/js/components/SignIn.js [hunk 7]
```diff
@@ -48,18 +97,31 @@ const SignIn = () => {
           <Typography component="h1" variant="h5">
             Sign in
           </Typography>
-          <Box component="form" noValidate sx={{ mt: 1 }}>
+          {
+            user_auth.errors === true ?
```
Comment (AubreyV) [inline resources/js/components/SignIn.js:101]: no need to elaborate the whole condition if the variable is boolean.

`variable_that_returns_a_boolean ? <what_to_do_when_true> : <what_to_do_when_false>`

File: resources/js/actions/userAuthActions.js (hunk 1 above)
Comment (michaelsilverio) [reply resources/js/actions/userAuthActions.js:14]: Made changes to userAuthActions.js. This is resolved.

File: resources/js/actions/userAuthActions.js (hunk 2 above)
Comment (michaelsilverio) [reply resources/js/actions/userAuthActions.js:24]: Used Prettier VS Code formatter to fix some code structures. This is resolved.

File: resources/js/actions/userAuthActions.js (hunk 3 above)
Comment (michaelsilverio) [reply resources/js/actions/userAuthActions.js:26]: Made changes to userAuthActons. This is resolved.

File: resources/js/actions/userAuthActions.js (hunk 4 above)
Comment (michaelsilverio) [reply resources/js/actions/userAuthActions.js:48]: Made changes to simplify the conditions of the validation functions. This is resolved

File: resources/js/actions/userAuthActions.js (hunk 5 above)
Comment (michaelsilverio) [reply resources/js/actions/userAuthActions.js:42]: I have read on the proper ways to make comments. Which is the first rule to Capitalize the first letter. I have made changes to all the comments based on the comments convention. Thank you for the suggestion ma'am. This is resolved.

File: resources/js/actions/userAuthActions.js (hunk 6 above)
Comment (michaelsilverio) [reply resources/js/actions/userAuthActions.js:66]: Made changes to userAuthActions.js. This is resolved.

File: resources/js/components/SignIn.js (hunk 7 above)
Comment (michaelsilverio) [reply resources/js/components/SignIn.js:101]: Shortened the condition in the ternary operation and removed unnecessary conditions. This is resolved

File: app/Http/Controllers/Auth/AuthController.php [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+class AuthController extends Controller
+{
+    public function login(LoginRequest $request)
+    {
+        $credentials = $request->only('email', 'password');
+        if (Auth::attempt($credentials)) 
+        {
+            return response()->json(['errors' => false]);
+        }
```
Comment (AubreyV) [inline app/Http/Controllers/Auth/AuthController.php:18]: We only place the `{` (open parenthesis) below if it's a class/function, not for conditional statements/loops

```suggestion
        if (Auth::attempt($credentials)) {
            return response()->json(['errors' => false]);
        }
```

File: app/Http/Controllers/Auth2/RegisterController.php [hunk 9]
```diff
@@ -0,0 +1,73 @@
+<?php
```
Comment (AubreyV) [inline app/Http/Controllers/Auth2/RegisterController.php:1]: Whats the difference of `Auth2` and `Auth` folders?

File: app/Http/Controllers/Auth/AuthController.php (hunk 8 above)
Comment (michaelsilverio) [reply app/Http/Controllers/Auth/AuthController.php:18]: Noted on this ma'am. Thank your for reminding.

File: app/Http/Controllers/Auth2/RegisterController.php (hunk 9 above)
Comment (michaelsilverio) [reply app/Http/Controllers/Auth2/RegisterController.php:1]: The Auth2 folder was actually a php artisan authentication scaffolding or auto generated Authentication which was originally named Auth folder. I was confused at that time if I were to use the generated authentication or make my I own authentication controllers. Which I eventually made my own controllers for the authentication and I recently delete the Auth2 folder. Sorry ma'am, for including the Auth2 folder in the commit .

**Review by AubreyV — APPROVED** [review]
AApproved functionality but code will be checked and fixed in another PR

**Pull request merged**